packages/devtools/lib/src/settings/settings_screen.dart

```diff
@@ -39,43 +44,141 @@ class SettingsScreen extends Screen {
           iconClass: 'octicon-gear masthead-item action-button active',
           showTab: false,
         ) {
-    _flagList = div(c: 'flag-list-container')
-      ..layoutVertical()
-      ..add(
-        CoreElement('h2', text: 'Dart VM Flag List'),
-      );
-    _sdkVersion = div(c: 'sdk-version-container')..flex(2);
-
     _controller = SettingsController(
       onFlagListChange: (FlagList flagList) {
         _flagList.add(flagList.flags.map((flag) => FlagDetailsUI(flag)));
       },
       onSdkVersionChange: (String sdkVersion) {
-        _sdkVersion.add(h2(text: sdkVersion));
+        _sdkVersion.text = sdkVersion;
       },
     );
   }
 
-  CoreElement _container;
   CoreElement _flagList;
-  CoreElement _screenDiv;
+
+  CoreElement _sdkVersionContainer;
+
   CoreElement _sdkVersion;
 
+  CoreElement _flutterVersionContainer;
+
+  CoreElement _versionContainer;
+
+  Spinner _loadingVersionSpinner;
+
   SettingsController _controller;
 
   @override
   CoreElement createContent(Framework framework) {
     ga_platform.setupDimensions();
     this.framework = framework;
-    _screenDiv = div(c: 'custom-scrollbar')..layoutVertical();
-    _container = div(c: 'settings-container');
-    _screenDiv
-      ..clear()
-      ..add(_container..clear());
-    _container.add(_sdkVersion);
-    _container.add(_flagList);
+
+    _initContent();
+    _checkForFlutterVersionService();
+
+    final CoreElement screenDiv = div(c: 'custom-scrollbar')..layoutVertical();
+
+    screenDiv
+      ..add([
+        div(c: 'section')
+          ..add([
+            h2(text: 'Version Information'),
+            _versionContainer,
+          ]),
+        div(c: 'section')
+          ..flex()
+          ..add([
+            h2(text: 'Dart VM Flag List'),
+            div(c: 'flag-list-container')
+              ..flex()
+              ..add(_flagList),
+          ])
+      ]);
 
     _controller.entering();
-    return _screenDiv;
+    return screenDiv;
+  }
+
+  void _initContent() {
+    _flagList = div(c: 'flag-list')..layoutVertical();
+
+    _versionContainer = div(c: 'version-container')
+      ..layoutVertical()
+      ..flex()
+      ..add([
+        _loadingVersionSpinner = Spinner.centered(),
+        _sdkVersionContainer = div()
+          ..layoutHorizontal()
+          ..add([
+            h5(text: 'Dart SDK:'),
+            _sdkVersion = h5(c: 'version-value'),
+          ])
+          ..hidden(true),
+        _flutterVersionContainer = div()
+          ..layoutVertical()
+          ..hidden(true)
+      ]);
+  }
+
+  void _checkForFlutterVersionService() async {
+    // Wait for a small delay to give DevTools a chance to pick up the
+    // registered service. This prevents the version UI from flickering.
+    await Future.delayed(const Duration(seconds: 1));
```

Comment:
we should avoid adding timeouts to the UI whenever possible. Instead of doing this, update the UI with the flutter version when it is available with a similar pattern to what we do to detect whether service extensions are available.